### PR TITLE
add verify tool golangci-lint, support staticcheck

### LINKED ISSUE #81518: Investigate golangci-lint to simplify our verify-* scripts
A lot of our go verification scripts are included in [golangci-lint](https://github.com/golangci/golangci-lint), which is designed to aggregate the tools for CI. It looks like the following are included:

- govet
- golint
- gofmt
- misspell
- staticcheck (https://github.com/kubernetes/kubernetes/pull/81189)

This tool might be able to simplify our verify scripts, as it encapsulates some of the functionality. Also, by aggregating the linters it may be much faster than running each tool independently. Finally, it would make adding new linters (supported by golangci-lint) much easier.

### PR DESCRIPTION
**Pull request by yylt**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
add golangci-lint tool , the tool aggregate the many tool , e.g govet , golint ,staticcheck

**Which issue(s) this PR fixes**:

Fixes #81518

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by BenTheElder** [comment]
Hi, thanks for your PR!

A couple of things:
- most likely these are going to broadly fail so we need a plan to mitigate that, eg https://github.com/kubernetes/kubernetes/blob/master/hack/.staticcheck_failures
- we should de-duplicate running staticcheck seperately (and possibly go lint, go vet, etc.)
- we need to ensure this actually runs in CI, check out `make verify`

/cc @tallclair

File: hack/.golangci.yml [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+  # large codebase. It's not practical to fix all existing issues at the moment
+  # of integration: much better don't allow issues in new code.
+  # Default is false.
+  new: false
+
+  # Show only new issues created after git revision `REV`
+  new-from-rev: REV
+
+  # Show only new issues created in git patch with set file path.
+  new-from-patch: ""
```
Comment (neolit123) [inline hack/.golangci.yml]: missing new line

File: build/tools.go [hunk 2]
```diff
@@ -38,4 +38,5 @@ import (
 	_ "k8s.io/gengo/examples/set-gen/generators"
 	_ "k8s.io/kube-openapi/cmd/openapi-gen"
 	_ "k8s.io/repo-infra/kazel"
+	_ "k8s.io/golangci/golangci-lint"
```
Comment (fejta) [inline build/tools.go]: This needs to get vendored in

**Comment by yylt** [comment]
Thack @BenTheElder   suggestion
    it is right to do as below?
    - [ ]  add basic shell script like `install-golangci-lint.sh`
    - [ ]  add verify-*.sh scritps which golangci-lint support, e.g. deadcode, varcheck ...

    when all done, we can use `make verify WHAT=deadcode` check.

    I am confused  what is actually runs in CI? , `make verify` will  run all verify-*.sh. So when name as verify-* , it will run in CI?

File: hack/verify-golangci-lint.sh [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+# Install tools we need, but only from vendor/...
+go install k8s.io/kubernetes/vendor/github.com/golangci/golangci-lint/cmd/golangci-lint
+
+# golangci-line check
+# All the skipping files are defined in hack/.spelling_failures
+for path in $(find ${KUBE_ROOT}/cmd -maxdepth 1 -type d) ;do
+    if [ ${path} = ${KUBE_ROOT}/cmd ] ;then
+        continue
+    fi
+    golangci-lint run -v ${path} -c ${KUBE_ROOT}/hack/.golangci.yml
```
Comment (tallclair) [inline hack/verify-golangci-lint.sh]: PR is missing the configuration file?

**Review by fejta — COMMENTED** [review]
/ok-to-test

File: hack/verify-golangci-lint.sh (hunk 3 above)
Comment (yylt) [reply hack/verify-golangci-lint.sh]: sorry, forget delete this file .

**Comment by yylt** [comment]
I will close this and open another PR

**Pull request closed without merging**